### ISSUE TITLE
Changes wont be saved

### ISSUE THREAD

**Issue by ikuV** [reflex-dev/reflex#3971]
Bug: A Save will not be changed after a hot reload or full reload of a reflex project

It worked a week ago and now following happens:

1. I change src
2. reflex updates the page
3. page still uses old source

Solutions i tried:
- clear borwser cache
- clear all pycaches
- try different reflex versions
- try different browsers
- delete .web folder

Example where i notived it:
```
class LoginState(rx.State):
    username: str = ""
    password: str = ""
    login_message: str = ""
    logged_in: bool = False
    login_red: bool = False

    def login(self):
        """Check the username and password."""
        if self.username == "Vuki" and self.password == "123":
            self.login_message = "Login successful!"
            self.logged_in = True
            return rx.redirect("/overview")
        else:
            self.login_message = "NEIN"
            return rx.redirect("/")

    def check_auth(self):
        if self.logged_in == False:
            return rx.redirect("/")
```

This is a very very basic auhentication script which works fine. I tried implementing firebase like in a youtube tutorial video.
Changes were not saved at all, still only the password "123" worked. When i now change the pass to "111" it still only accepts "123"

I hope you can follow.

Best wishes

Vuki

**Comment by ikuV** [comment]
Update: I found out it is a problem with the state itself. chaing the String of a label actually works. The State is not beeing saved correctly.

**Comment by ikuV** [comment]
Maybe i am doing sometinh wrong when using state handling? Have been searching for the past few hours. Have not found a solution.

**Comment by ikuV** [comment]
I recreated my project. Looks like there is an bug with state caching. I used the blank template and migrated my old files into the new project and it worked perfectly fine again.

Also no double reloads anymore.

**Comment by ikuV** [comment]
Issue still not fixxed... Getting tired of creating a new project and migrating files when states are not updated correctly.

**Comment by masenf** [comment]
@ikuV can you share your reflex, python, and OS versions that have been tried. I'm not able to reproduce the issue with the information given, but we definitely want to fix it.

**Comment by masenf** [comment]
Using this full repro code (based on your example above), i'm not able to reproduce the issue on 0.5.10, 0.6.0a4 or `main` using python3.11.

My methodology is to

* Run the app
* Load `/`
* Login with known bad credentials: `NEIN`
* Login with good credentials: redirect to `/overview` (success)
* Change the password hardcoded in the `login` event handler
* Wait for hot reload
* Load `/` again
* Login with new credentials: redirect to `/overview` (success)

```python
"""Welcome to Reflex! This file outlines the steps to create a basic app."""

import reflex as rx

class LoginState(rx.State):
    username: str = ""
    password: str = ""
    login_message: str = ""
    logged_in: bool = False
    login_red: bool = False

    def login(self):
        """Check the username and password."""
        if self.username == "Vuki" and self.password == "124":
            self.login_message = "Login successful!"
            self.logged_in = True
            return rx.redirect("/overview")
        else:
            self.login_message = "NEIN"
            return rx.redirect("/")

    def check_auth(self):
        if self.logged_in == False:
            return rx.redirect("/")

def index() -> rx.Component:
    return rx.container(
        rx.text(LoginState.login_message),
        rx.form(
            rx.input(name="username", label="Username", on_change=LoginState.set_username),
            rx.input(name="password", type="password", label="Password", on_change=LoginState.set_password),
            rx.button("Login", on_click=LoginState.login),
            on_submit=rx.console_log("Form submitted"),
        ),
    )

def overview() -> rx.Component:
    return rx.container(
        rx.heading("Overview"),
        rx.text(f"You made it {LoginState.username}! {LoginState.login_message}"),
    )

app = rx.App()
app.add_page(index)
app.add_page(overview)
```

Have you tried opening `/` in a new tab _after_ the hot reload and see if the "old code" is still being used on the new tab?

**Comment by Lendemor** [comment]
@ikuV This should be fixed in 0.6.1 (next release), please reopen the issue with a code snippet for reproducing it if it's still happening after.
